src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCM.java

```diff
@@ -523,33 +342,54 @@ public boolean getShowGitToolOptions() {
          * @param req      request
          * @param formData json data
          * @return a new BitbucketSCM instance
-         * @throws FormException if any data is missing
          */
         @Override
         public SCM newInstance(@Nullable StaplerRequest req, JSONObject formData) throws FormException {
-            BitbucketSCM scm = (BitbucketSCM) super.newInstance(req, formData);
-            scm.setBitbucketClientFactoryProvider(bitbucketClientFactoryProvider);
-            scm.setBitbucketPluginConfiguration(bitbucketPluginConfiguration);
-
             String serverId = formData.getString("serverId");
-            if (isBlank(serverId)) {
-                LOGGER.info("Error creating the Bitbucket SCM: The serverId cannot be blank");
-                return scm;
-            }
-            Optional<BitbucketServerConfiguration> maybeServerConf = bitbucketPluginConfiguration.getServerById(serverId);
-            if (!maybeServerConf.isPresent()) {
-                LOGGER.info("Error creating the Bitbucket SCM: No server configuration for the given server id " + serverId);
-                return scm;
-            }
-            BitbucketServerConfiguration serverConf = maybeServerConf.get();
+            bitbucketPluginConfiguration.getServerById(serverId).ifPresent(serverConf -> {
+                String credentialsId = stripToNull(formData.getString("credentialsId"));
+                if (credentialsId == null) {
+                    credentialsId = serverConf.getCredentialsId();
+                    formData.put("credentialsId", credentialsId);
+                }
+                BitbucketCredentials credentials =
+                        jenkinsToBitbucketCredentials.toBitbucketCredentials(credentialsId, serverConf);
+                BitbucketClientFactory clientFactory =
+                        bitbucketClientFactoryProvider.getClient(serverConf.getBaseUrl(), credentials);
+
+                String projectName = formData.getString("projectName");
+                String repositoryName = formData.getString("repositoryName");
+                BitbucketProject project = getBitbucketProject(projectName, clientFactory);
+                formData.put("projectKey", project.getKey());
+                BitbucketRepository repository =
+                        getBitbucketRepository(projectName, repositoryName, clientFactory, project);
+                formData.put("repositorySlug", repository.getSlug());
+
+                String existingMirrorSelection = formData.getString("mirrorName");
+                String cloneUrl;
+                if (!isEmpty(existingMirrorSelection)) {
+                    EnrichedBitbucketMirroredRepository mirroredRepo = createMirrorHandler().fetchRepository(
```

Comment:
This may throw, we should probably catch, log and not blow up completely?

Reply:
Fixed